crates/core/src/db/datastore/locking_tx_datastore/btree_index.rs

```diff
@@ -57,28 +58,33 @@ impl Iterator for BTreeIndexRangeIter<'_> {
 pub(crate) struct BTreeIndex {
     pub(crate) index_id: IndexId,
     pub(crate) table_id: u32,
-    pub(crate) col_id: u32,
+    pub(crate) cols: NonEmpty<u32>,
     pub(crate) name: String,
     pub(crate) is_unique: bool,
     idx: BTreeSet<IndexKey>,
```

Comment:
Ok, the major reason for this is because "single columns" are the most common case for a index key, and "multiple columns" is relatively rare.

So, when storing the key, most of the time will be like:

```sql

CREATE a INDEX a ON t1 (column_i32); -- Most common case index `BuiltIn`

CREATE a INDEX b ON t1 (column_i32, column_i32); -- Sometimes we have more complex keys

```

Will be on memory:

```rust
[AlgebraicValue::I32(0)] = Row 
[AlgebraicValue::Product(AlgebraicValue::I32(0), AlgebraicValue::I32(1))] = Row 
```

Reply:
Yes, is here: https://github.com/clockworklabs/SpacetimeDB/pull/243/files#diff-81f8996ebd80b25ce7aa1420ae414b31d532b7dfc46d9aee281b753398b13e2eR77